### ISSUE TITLE
ultheme and writegood not detected when installed as system dependencies

### ISSUE THREAD

**Issue by yochananmarqos** [kmwallio/ThiefMD#52]
As I mentioned on your Reddit post, I just added ThiefMD to the AUR: [`thiefmd`](https://aur.archlinux.org/packages/thiefmd/). It detects GXml 0.18.2 installed as a system dependency, but not `ultheme-vala` and `libwritegood-vala` when installed from the latest commit.

```
Run-time dependency ultheme-0.1 found: NO (tried pkgconfig and cmake)
```
```
Run-time dependency writegood-0.1 found: NO (tried pkgconfig and cmake)
```

**Comment by kmwallio** [comment]
Is it possible to point it at the [release tar.gz](https://github.com/kmwallio/ThiefMD/releases/tag/v0.0.12-luminance)?  If it helps to have it in a fixed location for future updates, I can put it on S3 or some other file server.

**Comment by yochananmarqos** [comment]
Huh? Point what and who and do with with it? I think you misunderstood. There's no issue with the release tarball.

I'm aiming to build the package using system dependencies instead of pre-built libraries via the Meson Subprojects.

Using subprojects is [discouraged](This is usually frowned upon because it makes it hard to add these kinds of projects into e.g. those Linux distributions that forbid bundled libraries.) on platforms with a native packaging system, like Arch.

> This is usually frowned upon because it makes it hard to add these kinds of projects into e.g. those Linux distributions that forbid bundled libraries.

It's great for Flatpaks, but not building an AUR package. I was wondering why it can't detect the aforementioned dependencies when installed.

**Comment by kmwallio** [comment]
Oh, I may have an issue in the meson.build file for not falling back to the OS libraries.  I'll check.

**Comment by yochananmarqos** [comment]
Well, it did detect GXml. That's a start. I haven't tried `--wrap-mode=nofallback`, but it fails with `--wrap-mode nodownload`. The latter is all I should need.

**Comment by kmwallio** [comment]
Can you see if ceab3a333c126b50d6273edc7bde345facf255d0 fixes it?

**Comment by yochananmarqos** [comment]
Well, it did detect them as dependencies. However, it failed building. That's curious as the project builds fine using the subprojects.

```
+ exec meson setup --prefix /usr --libexecdir lib --sbindir bin --buildtype plain --auto-features enabled --wrap-mode nodownload -D b_lto=true -D b_pie=true ThiefMD build
The Meson build system
Version: 0.55.3
Source dir: /home/yochanan/Documents/AUR/thiefmd/src/ThiefMD
Build dir: /home/yochanan/Documents/AUR/thiefmd/src/build
Build type: native build
Project name: com.github.kmwallio.thiefmd
Project version: 0.0.12
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
C compiler for the host machine: cc (gcc 10.2.0 "cc (GCC) 10.2.0")
C linker for the host machine: cc ld.bfd 2.35
Vala compiler for the host machine: valac (valac 0.50.1)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/sbin/pkg-config (1.7.3)
Configuring config.h using configuration
Found pkg-config: /usr/sbin/pkg-config (1.7.3)
Run-time dependency libmarkdown found: YES 2.2.7
Run-time dependency ultheme found: YES 0.0.1
Run-time dependency writegood found: YES 0.0.1
Run-time dependency gobject-2.0 found: YES 2.66.1
Run-time dependency gtksourceview-3.0 found: YES 3.24.12
Run-time dependency gtk+-3.0 found: YES 3.24.23
Run-time dependency gtkspell3-3.0 found: YES 3.0.10
Run-time dependency webkit2gtk-4.0 found: YES 2.30.1
Run-time dependency clutter-1.0 found: YES 1.26.4
Run-time dependency json-glib-1.0 found: YES 1.6.0
Run-time dependency gee-0.8 found: YES 0.20.3
Program meson/post_install.py found: YES (/usr/bin/python /home/yochanan/Documents/AUR/thiefmd/src/ThiefMD/meson/post_install.py)
Program glib-compile-schemas found: YES
Build targets in project: 4

Found ninja-1.10.1 at /usr/sbin/ninja
Found runner: ['/usr/sbin/ninja']
ninja: Entering directory `build'
[4/33] Compiling Vala source ../ThiefMD/src/....vala ../ThiefMD/src/Controllers/Pandoc.vala
warning: Ignoring source file `/home/yochanan/Documents/AUR/thiefmd/src/ThiefMD/vapi/libmarkdown.vapi', which was already added to this context
../ThiefMD/src/Widgets/HeaderBar.vala:28.9-28.40: warning: field `ThiefMD.Widgets.Headerbar.search_button' never used
        private Gtk.Button search_button;
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../ThiefMD/src/Widgets/Sheets.vala:191.17-191.20: warning: local variable `keys' declared but never used
            var keys = _sheets.keys;
                ^^^^
../ThiefMD/src/Widgets/Sheets.vala:275.15-275.29: warning: unreachable catch clause detected
            } catch (Error e) {
              ^^^^^^^^^^^^^^^
../ThiefMD/src/Widgets/Editor.vala:571.23-571.23: warning: local variable `m' declared but never used
            int w, h, m, p;
                      ^
../ThiefMD/src/Widgets/Editor.vala:571.26-571.26: warning: local variable `p' declared but never used
            int w, h, m, p;
                         ^
../ThiefMD/src/Widgets/Editor.vala:739.29-739.41: warning: local variable `settings_menu' declared but never used
                        var settings_menu = AppSettings.get_default ();
                            ^^^^^^^^^^^^^
../ThiefMD/src/Widgets/Library.vala:48.9-48.31: warning: method `ThiefMD.Widgets.LibPair.to_string' never used
        public string to_string () {
        ^^^^^^^^^^^^^^^^^^^^^^^
../ThiefMD/src/Widgets/Preview.vala:112.17-112.29: warning: switch does not handle `NAVIGATION_ACTION' of enum `WebKit.PolicyDecisionType'
                switch (type) {
                ^^^^^^^^^^^^^
../ThiefMD/src/Widgets/PreviewWindow.vala:55.23-55.23: warning: local variable `m' declared but never used
            int w, h, m, p;
                      ^
../ThiefMD/src/Widgets/PreviewWindow.vala:55.26-55.26: warning: local variable `p' declared but never used
            int w, h, m, p;
                         ^
../ThiefMD/src/Widgets/PublisherPreviewWindow.vala:39.23-39.23: warning: local variable `m' declared but never used
            int w, h, m, p;
                      ^
../ThiefMD/src/Widgets/PublisherPreviewWindow.vala:39.26-39.26: warning: local variable `p' declared but never used
            int w, h, m, p;
                         ^
../ThiefMD/src/Widgets/KeyBindings.vala:72.27-72.41: warning: unreachable catch clause detected
                        } catch (Error e) {
                          ^^^^^^^^^^^^^^^
../ThiefMD/src/Controllers/FileManager.vala:65.13-65.16: warning: local variable `lock' declared but never used
        var lock = new FileLock ();
            ^^^^
../ThiefMD/src/Controllers/FileManager.vala:68.16-68.19: warning: local variable `text' declared but never used
        string text;
               ^^^^
../ThiefMD/src/Controllers/FileManager.vala:63.46-63.70: warning: use of possibly unassigned parameter `editor'
    public void open_file (string file_path, out Widgets.Editor editor) {
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
../ThiefMD/src/Controllers/FileManager.vala:321.13-321.16: warning: local variable `lock' declared but never used
        var lock = new FileLock ();
            ^^^^
../ThiefMD/src/Controllers/FileManager.vala:367.13-367.16: warning: local variable `lock' declared but never used
        var lock = new FileLock ();
            ^^^^
../ThiefMD/src/Controllers/SheetManager.vala:122.5-122.44: warning: field `ThiefMD.Controllers.SheetManager.sheet_worker_thread' never used
    private Thread<bool> sheet_worker_thread;
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../ThiefMD/src/Controllers/SheetManager.vala:129.5-129.31: warning: method `ThiefMD.Controllers.SheetManager.preload_sheets' never used
    private bool preload_sheets () {
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../ThiefMD/src/Controllers/UI.vala:528.23-528.30: warning: copying delegates is not supported
Compilation succeeded - 22 warning(s)
[33/33] Linking target com.github.kmwallio.thiefmd
FAILED: com.github.kmwallio.thiefmd 
cc  -o com.github.kmwallio.thiefmd com.github.kmwallio.thiefmd.p/meson-generated_.._as-resources.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Application.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Constants_AppSettings.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Constants_DragAndDrop.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Constants_ThiefProperties.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Constants_Helpers.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_About.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_HeaderBar.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Sheets.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Sheet.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Editor.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Library.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_NewFolder.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_NewSheet.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Preferences.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_Preview.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_PreviewWindow.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_PublisherPreviewWindow.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_KeyBindings.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_QuickPreferences.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_ThemeSelector.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_DefaultTheme.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Widgets_ThemePreview.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_FileManager.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_Dialogs.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_SheetManager.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_UI.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_UserData.c.o com.github.kmwallio.thiefmd.p/meson-generated_src_Controllers_Pandoc.c.o -flto -Wl,--as-needed -Wl,--no-undefined -pie -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wl,--start-group /usr/lib/libgobject-2.0.so /usr/lib/libglib-2.0.so /usr/lib/libgtksourceview-3.0.so /usr/lib/libgtk-3.so /usr/lib/libgdk-3.so /usr/lib/libz.so /usr/lib/libpangocairo-1.0.so /usr/lib/libpango-1.0.so /usr/lib/libharfbuzz.so /usr/lib/libatk-1.0.so /usr/lib/libcairo-gobject.so /usr/lib/libcairo.so /usr/lib/libgdk_pixbuf-2.0.so /usr/lib/libgio-2.0.so /usr/lib/libgtkspell3-3.so /usr/lib/libenchant-2.so /usr/lib/libwebkit2gtk-4.0.so /usr/lib/libsoup-2.4.so /usr/lib/libjavascriptcoregtk-4.0.so /usr/lib/libclutter-1.0.so /usr/lib/libcogl-path.so /usr/lib/libcogl-pango.so /usr/lib/libcogl.so /usr/lib/libgmodule-2.0.so -pthread /usr/lib/libwayland-egl.so /usr/lib/libgbm.so /usr/lib/libdrm.so /usr/lib/libEGL.so /usr/lib/libXrandr.so /usr/lib/libjson-glib-1.0.so /usr/lib/libwayland-cursor.so /usr/lib/libwayland-client.so /usr/lib/libxkbcommon.so /usr/lib/libwayland-server.so /usr/lib/libX11.so /usr/lib/libXext.so /usr/lib/libXdamage.so /usr/lib/libXfixes.so /usr/lib/libXcomposite.so /usr/lib/libXi.so /usr/lib/libgee-0.8.so /usr/lib/libultheme.a /usr/lib/libarchive.so /usr/lib/libxml2.so /usr/lib/libwritegood.a /usr/lib/libmarkdown.so -Wl,--end-group
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `______lambda7__gsource_func':
<artificial>:(.text+0x7f2): undefined reference to `write_good_checker_recheck_all'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `thief_md_widgets_editor_write_good_recheck':
<artificial>:(.text+0x852): undefined reference to `write_good_checker_recheck_all'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `thief_md_widgets_editor_update_settings':
<artificial>:(.text+0x99fc): undefined reference to `write_good_checker_attach'
/usr/sbin/ld: <artificial>:(.text+0x9c8d): undefined reference to `write_good_checker_detach'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `thief_md_widgets_editor_set_am_active':
<artificial>:(.text+0xa744): undefined reference to `write_good_checker_attach'
/usr/sbin/ld: <artificial>:(.text+0xa7d1): undefined reference to `write_good_checker_detach'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `thief_md_widgets_editor_constructor':
<artificial>:(.text+0xaf43): undefined reference to `write_good_checker_new'
/usr/sbin/ld: <artificial>:(.text+0xaf55): undefined reference to `write_good_checker_unref'
/usr/sbin/ld: <artificial>:(.text+0xafcf): undefined reference to `write_good_checker_attach'
/usr/sbin/ld: <artificial>:(.text+0xb124): undefined reference to `write_good_checker_detach'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans1.ltrans.o: in function `thief_md_widgets_editor_finalize':
<artificial>:(.text+0xb69d): undefined reference to `write_good_checker_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `thief_md_widgets_theme_preview_finalize':
<artificial>:(.text+0x297b): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: <artificial>:(.text+0x29cb): undefined reference to `ultheme_hex_color_palette_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `thief_md_widgets_theme_preview_set_preview_state':
<artificial>:(.text+0x2a4a): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x2a85): undefined reference to `ultheme_parser_get_light_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x2b41): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `___lambda61__gtk_button_clicked':
<artificial>:(.text+0x2c5c): undefined reference to `ultheme_parser_get_light_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x2c83): undefined reference to `ultheme_parser_get_light_theme_palette'
/usr/sbin/ld: <artificial>:(.text+0x2c96): undefined reference to `ultheme_hex_color_palette_unref'
/usr/sbin/ld: <artificial>:(.text+0x2d0f): undefined reference to `ultheme_parser_base_file_name'
/usr/sbin/ld: <artificial>:(.text+0x2d6a): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x2d91): undefined reference to `ultheme_parser_get_dark_theme_palette'
/usr/sbin/ld: <artificial>:(.text+0x2e0c): undefined reference to `ultheme_parser_base_file_name'
/usr/sbin/ld: <artificial>:(.text+0x2e7e): undefined reference to `ultheme_parser_base_file_name'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `thief_md_widgets_theme_preview_construct':
<artificial>:(.text+0x47f4): undefined reference to `ultheme_parser_ref'
/usr/sbin/ld: <artificial>:(.text+0x480a): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: <artificial>:(.text+0x4928): undefined reference to `ultheme_parser_get_theme_name'
/usr/sbin/ld: <artificial>:(.text+0x499b): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x4a12): undefined reference to `ultheme_parser_get_dark_theme'
/usr/sbin/ld: <artificial>:(.text+0x4a6a): undefined reference to `ultheme_parser_get_light_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x4ae1): undefined reference to `ultheme_parser_get_light_theme'
/usr/sbin/ld: <artificial>:(.text+0x4c02): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x4cc2): undefined reference to `ultheme_parser_get_light_theme_id'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `thief_md_widgets_preferences_construct':
<artificial>:(.text+0x68cd): undefined reference to `ultheme_parser_ref'
/usr/sbin/ld: <artificial>:(.text+0x6968): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans2.ltrans.o: in function `_thief_md_widgets_theme_selector_preview_drop_on_drag_data_received_gtk_widget_drag_data_received':
<artificial>:(.text+0x78d8): undefined reference to `ultheme_parser_new'
/usr/sbin/ld: <artificial>:(.text+0x79a0): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans3.ltrans.o: in function `_ultheme_parser_unref0_.lto_priv.0':
<artificial>:(.text+0x777): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans3.ltrans.o: in function `thief_md_controllers_ui_add_user_theme':
<artificial>:(.text+0x1b7b): undefined reference to `ultheme_parser_ref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans3.ltrans.o: in function `_thief_md_controllers_ui_load_themes_gthread_func.lto_priv.0':
<artificial>:(.text+0x1d6f): undefined reference to `ultheme_parser_new'
/usr/sbin/ld: <artificial>:(.text+0x1dd9): undefined reference to `ultheme_parser_get_dark_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x1e3f): undefined reference to `ultheme_parser_get_dark_theme'
/usr/sbin/ld: <artificial>:(.text+0x1ebe): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: <artificial>:(.text+0x1ef1): undefined reference to `ultheme_parser_get_light_theme_id'
/usr/sbin/ld: <artificial>:(.text+0x1f58): undefined reference to `ultheme_parser_get_light_theme'
/usr/sbin/ld: <artificial>:(.text+0x20d8): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans3.ltrans.o: in function `thief_md_controllers_ui_set_css_scheme':
<artificial>:(.text+0x22cc): undefined reference to `ultheme_hex_color_palette_get_global'
/usr/sbin/ld: <artificial>:(.text+0x22d5): undefined reference to `ultheme_hex_color_pair_get_background'
/usr/sbin/ld: <artificial>:(.text+0x22e1): undefined reference to `ultheme_hex_color_palette_get_global_active'
/usr/sbin/ld: <artificial>:(.text+0x22ea): undefined reference to `ultheme_hex_color_pair_get_background'
/usr/sbin/ld: <artificial>:(.text+0x22f6): undefined reference to `ultheme_hex_color_palette_get_headers'
/usr/sbin/ld: <artificial>:(.text+0x22ff): undefined reference to `ultheme_hex_color_pair_get_foreground'
/usr/sbin/ld: <artificial>:(.text+0x230b): undefined reference to `ultheme_hex_color_palette_get_global_active'
/usr/sbin/ld: <artificial>:(.text+0x2314): undefined reference to `ultheme_hex_color_pair_get_foreground'
/usr/sbin/ld: <artificial>:(.text+0x2320): undefined reference to `ultheme_hex_color_palette_get_global'
/usr/sbin/ld: <artificial>:(.text+0x2329): undefined reference to `ultheme_hex_color_pair_get_foreground'
/usr/sbin/ld: <artificial>:(.text+0x24cb): undefined reference to `ultheme_hex_color_palette_get_global'
/usr/sbin/ld: <artificial>:(.text+0x24d4): undefined reference to `ultheme_hex_color_pair_get_background'
/usr/sbin/ld: /tmp/com.github.kmwallio.thiefmd.79GDNU.ltrans3.ltrans.o: in function `thief_md_controllers_ui_load_css_scheme':
<artificial>:(.text+0x2a5d): undefined reference to `ultheme_parser_new'
/usr/sbin/ld: <artificial>:(.text+0x2acd): undefined reference to `ultheme_hex_color_palette_unref'
/usr/sbin/ld: <artificial>:(.text+0x2c16): undefined reference to `ultheme_parser_get_dark_theme_palette'
/usr/sbin/ld: <artificial>:(.text+0x2c35): undefined reference to `ultheme_parser_unref'
/usr/sbin/ld: <artificial>:(.text+0x2c65): undefined reference to `ultheme_hex_color_palette_unref'
/usr/sbin/ld: <artificial>:(.text+0x2c83): undefined reference to `ultheme_parser_get_light_theme_palette'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by kmwallio** [comment]
I was able to repro something similar, but rebuilding ultheme and writegood, then doing a ninja install from the repos fixed it for me.  I'll see if I can get Arch setup in a VM.

Thanks for helping and working on this!

**Comment by kmwallio** [comment]
@yochananmarqos, if I include them as git submodules and compile the libraries statically into the application, will that fix the issue for you? It'll also prevent conflicts if other developers use and ship the library from the version embedded in ThiefMD.

**Comment by yochananmarqos** [comment]
No, that's the same thing. Perhaps take a look at [GameMode](https://github.com/FeralInteractive/gamemode), I just went through a similar thing along with the Gentoo package maintainer before it finally got added to the Arch community repo.

**Comment by yochananmarqos** [comment]
Now it's neither looking for `ultheme` & `writegood` nor fetching them with 0.1.0:

```
The Meson build system
Version: 0.55.3
Source dir: /home/yochanan/Documents/AUR/thiefmd/src/ThiefMD-0.1.0-inandout
Build dir: /home/yochanan/Documents/AUR/thiefmd/src/build
Build type: native build
Project name: com.github.kmwallio.thiefmd
Project version: 0.1.0
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
Using 'CFLAGS' from environment with value: '-march=x86-64 -mtune=generic -O2 -pipe -fno-plt'
Using 'LDFLAGS' from environment with value: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
Using 'CPPFLAGS' from environment with value: '-D_FORTIFY_SOURCE=2'
C compiler for the host machine: cc (gcc 10.2.0 "cc (GCC) 10.2.0")
C linker for the host machine: cc ld.bfd 2.35
Vala compiler for the host machine: valac (valac 0.50.1)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/sbin/pkg-config (1.7.3)
Configuring config.h using configuration
Found pkg-config: /usr/sbin/pkg-config (1.7.3)
Run-time dependency libmarkdown found: YES 2.2.7
Run-time dependency gobject-2.0 found: YES 2.66.1
Run-time dependency gtksourceview-4 found: YES 4.8.0
Run-time dependency gtk+-3.0 found: YES 3.24.23
Run-time dependency gtkspell3-3.0 found: YES 3.0.10
Run-time dependency webkit2gtk-4.0 found: YES 2.30.1
Run-time dependency clutter-1.0 found: YES 1.26.4
Run-time dependency json-glib-1.0 found: YES 1.6.0
Run-time dependency gee-0.8 found: YES 0.20.3
Run-time dependency libarchive found: YES 3.4.3
Run-time dependency libxml-2.0 found: YES 2.9.10

ThiefMD-0.1.0-inandout/meson.build:55:0: ERROR: File src/ultheme/src/HexColors.vala does not exist.
```

**Comment by kmwallio** [comment]
yeah, I wound up switching to git submodules to see if I could fix the elementryOS build since it's not compatible with wraps.

I'll have to have add some flags depending on where it's building.

**Comment by yochananmarqos** [comment]
Ah, of course. I've been a bit distracted this evening. It builds fine pulling the submodules. Actually, that is better as there are no longer any static libs.

**Comment by kmwallio** [comment]
Sorry for the delay, I finally got around to setting up Arch in a VM.

I just tried it following [these instructions](https://wiki.archlinux.org/index.php/Arch_User_Repository#Installing_and_upgrading_packages) and it worked.

Thank you so much! I was going to update the readme and website with instructions. Do most people use yay or some other AUR helper?

Thanks again, it was super exciting to see everything compile and work. Feel free to let me know if there's anything I have to do on my end.
